datalad/interface/tests/test_run_procedure.py

```diff
@@ -86,3 +126,144 @@ def test_procedure_discovery():
              'path' in p
              for p in ps]),
         len(ps))
+
+    # set up dataset with registered procedure (c&p from test_basics):
+    ds = Dataset(path).create(force=True)
+    # TODO: this procedure would leave a clean dataset, but `run` cannot handle dirty
+    # input yet, so manual for now
+    # V6FACT: this leaves the file staged, but not committed
+    ds.add('code', to_git=True)
+    # V6FACT: even this leaves it staged
+    ds.add('.')
+    # V6FACT: but this finally commits it
+    ds.save()
+    # TODO remove above two lines
```

Comment:
Again, I know this is just copied, and I don't think it needs to be addressed in this PR, but based on quick testing, v6 mode now appears to behave as expected without these lines (presumably either due to an update in annex or a fix on our end).

Reply:
Well, I should probably try to remove it within this PR. It will hang around there for quite some while otherwise, I guess.

Reply:
> It will hang around there for quite some while otherwise, I guess.

I've noted it, so I would look into it once this PR lands, but of course I don't mind you looking into it now.